apps/wallet/src/background/Tabs.ts

```diff
@@ -155,22 +155,58 @@ class Tabs {
         return this._onActiveOrigin.asObservable();
     }
 
-    public async highlight(option: { url: string } | { tabID: number }) {
+    public async highlight(
+        option: { windowID?: number } & (
+            | {
+                  url: string;
+                  match?: (values: {
+                      url: string;
+                      inAppRedirectUrl?: string;
+                  }) => boolean;
+              }
+            | { tabID: number }
+        )
+    ) {
         let tabToHighlight: BrowserTabs.Tab | null = null;
         if ('tabID' in option) {
             try {
                 tabToHighlight = await Browser.tabs.get(option.tabID);
             } catch (e) {
                 //Do nothing
             }
-        }
-        if ('url' in option) {
-            const url = option.url.split('#')[0];
+        } else {
+            const inAppUrlToMatch = option.url.split('#')[1] || '';
             const tabs = (
                 await Browser.tabs.query({
-                    url,
+                    url: option.url.split('#')[0],
+                    windowId: option.windowID,
                 })
-            ).filter((aTab) => aTab.url === option.url);
+            ).filter((aTab) => {
+                let inAppRedirectUrl: string | undefined = undefined;
+                if (aTab.url === option.url) {
+                    return true;
+                }
+                try {
+                    const tabURL = new URL(aTab.url || '');
+                    if (tabURL.hash.startsWith('#/locked?url=')) {
+                        inAppRedirectUrl = decodeURIComponent(
+                            tabURL.hash.replace('#/locked?url=', '')
+                        );
+                        if (inAppRedirectUrl === inAppUrlToMatch) {
+                            return true;
+                        }
+                    }
+                } catch (e) {
+                    // do nothing
```

Comment:
nit: instead of having the empty catch here and defaulting `aTab.url` to an empty string here and below, would it make sense to just stick a truthy check at the top of the filter code?

```
if (!aTab.url) {
  return false;
}
```

Reply:
Yeah good point. But I think I will keep the empty catch in case URL throws though. (Probably url should be always valid but just in case)